### PR TITLE
Refs #50 - support for NaN and +/- Inf.

### PR DESCRIPTION
**Pull request by TreyE**

### CODE REVIEW

File: test/bson_test.exs [hunk 1]
```diff
@@ -141,7 +141,43 @@ defmodule BSONTest do
...
+
+  test "decode float negative Infinity" do
+    assert decode(@binNegInf) == @mapNegInf
+  end
+
+  test "encode float negative Infinity" do
+    assert encode(@mapNegInf) == @binNegInf
+  end
+
+
```
Comment (ericmj) [inline test/bson_test.exs]: ✂️ whitespace

File: test/bson_test.exs [hunk 2]
```diff
@@ -141,7 +141,43 @@ defmodule BSONTest do
...
   defp encode(value) do
     value |> BSON.encode |> IO.iodata_to_binary
   end
+
+
```
Comment (ericmj) [inline test/bson_test.exs]: ✂️ whitespace

**Comment by TreyE** [comment]
#50 I've fixed the whitespace issues.

**Comment by ericmj** [comment]
There's still an extra line on both places unfortunately.

**Comment by TreyE** [comment]
Not sure I understand your spacing guidelines, is it that there should be no space between the end of a final method in a module and the module end?

In other words, should https://github.com/TreyE/mongodb/blob/2506a17a701e59177cbb9bbba7b1f86d27eed846/test/bson_test.exs#L180 not be there?

**Comment by TreyE** [comment]
Going to remove this and do a clean pull request.

**Pull request closed without merging**